pyalarmdotcomajax/pyalarmdotcomajax.py

```diff
@@ -205,13 +227,45 @@ async def async_login(self):
             return False
         return await self._async_get_system_info()
 
+    async def async_update_lock(self):
```

Comment:
Why do we need this as a standalone method? Isn't it just done in `async_update`?

Reply:
I was not too sure about this, but ended up creating a new method, because:

1) The fetching of the state have a bit different code. Sure, could work around.
2) The booleans force_bypass, no_entry_delay and silent_arming did not make sense in a lock context, so I decided to just create a new method and remove this from the lock functionality (see also alarmdotcom pull request)

Reply:
I see, I didn't realize this was actually supporting a different HA platform. There seems to be a lot of code that can be reused between the two functions. Can you refactor it a bit so that `async_update_lock` and `async_update` use another function which has the common code?

Reply:
Good idea. Will get around to it, when I have the time.

Reply:
I decided to go back to use the same function, but with an argument alarm or lock to make it behave different